jobs/payment-jobs/tasks/stale_payment_task.py

```diff
@@ -89,3 +97,40 @@ def _delete_marked_payments(cls):
             except BusinessException as err:  # just catch and continue .Don't stop
                 current_app.logger.warn('Error on delete_payment')
                 current_app.logger.warn(err)
+
+    @classmethod
+    def _verify_recent_payments(cls):
+        """Verify recent invoice with PAYBC."""
+        recent_invoices = InvoiceModel.find_recent_cc_invoices(days=2)
+        if len(recent_invoices) == 0:
+            current_app.logger.info(f'Verify Invoice Job Ran at {datetime.datetime.now(tz=datetime.timezone.utc)}.'
+                                    'But No records found!')
+        for invoice in recent_invoices:
+            try:
+                current_app.logger.info(f'Verify Invoice Job found records.Invoice Id: {invoice.id}')
+                paybc_invoice = DirectPayService.query_order_status(invoice, InvoiceReferenceStatus.ACTIVE.value)
+                if paybc_invoice.get('paymentstatus') in STATUS_PAID:
+                    invoice_reference = InvoiceReference.find_active_reference_by_invoice_id(invoice.id)
```

Comment:
Is it possible to have an invoice without an invoice reference? I think no for DIRECT_PAY